File: py/1179.py
even, odd = [], []
count = 15
while count > 0:
    n = int(input())
    
    if n % 2 == 0:
        even.append(n)
    else:
        odd.append(n)
        
    if len(even) == 5:
        for i in range(len(even)):
            print(f'par[{i}] = {even[i]}')
        
        even = []
        
    if len(odd) == 5:
        for i in range(len(odd)):
            print(f'impar[{i}] = {odd[i]}')
        
        odd = []
        
    count -= 1
    
for i in range(len(odd)):
    print(f'impar[{i}] = {odd[i]}')    

for i in range(len(even)):
    print(f'par[{i}] = {even[i]}')

File: cpp/1179.cpp
#include <iostream>
#include <stdio.h>
#include <vector>

using namespace std;

int main()
{
    vector<int> odd;
    vector<int> even;
    int n, i, j, len_odd, len_even;

    for (i = 0; i < 15; i++) {
        cin >> n;

        if (n % 2 == 0) {
            even.push_back(n);
        } else {
            odd.push_back(n);
        }

        len_odd = odd.size();
        len_even = even.size();

        if (len_odd == 5) {
            for (j = 0; j < len_odd; j++) {
                printf("impar[%d] = %d\n", j, odd.at(j));
            }

            odd.clear();
        }

        if (len_even == 5) {
            for (j = 0; j < len_even; j++) {
                printf("par[%d] = %d\n", j, even.at(j));
            }

            even.clear();
        }
    }

    for (j = 0; j < len_odd; j++) {
        printf("impar[%d] = %d\n", j, odd.at(j));
    }

    for (j = 0; j < len_even; j++) {
        printf("par[%d] = %d\n", j, even.at(j));
    }

    return 0;
}